src/libraries/System.IO.Compression/tests/ZipArchive/zip_UpdateTests.Comments.cs

```diff
@@ -11,179 +11,86 @@ namespace System.IO.Compression.Tests
-        [InlineData(65001, 20127)] // utf8, ascii
-        // General purpose bit flag must get the appropriate bit set if a file comment or an entry name is unicode
-        public static void Update_Comments_EntryName_DifferentEncodings(int entryNameCodePage, int commentCodePage)
-        {
-            Encoding entryNameEncoding = Encoding.GetEncoding(entryNameCodePage);
-            Encoding commentEncoding = Encoding.GetEncoding(commentCodePage);
+        [MemberData(nameof(Latin1Comment_Data))]
+        public static void Update_Comment_Utf8EntryName_Latin1Encoding(string originalComment, string expectedComment) =>
+            // Emoji not supported by latin1
+            Update_Comment_EntryName_Encoding_Internal($"{LowerCaseOUmlautChar}.txt",
+                originalComment, expectedComment, Encoding.Latin1,
+                new string('a', ushort.MaxValue - 1) + $"{CopyrightChar}{CopyrightChar}", new string('a', ushort.MaxValue - 1) + $"{CopyrightChar}");
 
-            string entryName = entryNameEncoding.GetString(Encoding.ASCII.GetBytes("EntryName.txt"));
-            string comment = commentEncoding.GetString(Encoding.ASCII.GetBytes("Comment!"));
 
-            var stream = new MemoryStream();
-            var testStream = new WrappedStream(stream, true, true, true, null);
+        private static void Update_Comment_EntryName_Encoding_Internal(string entryName,
+            string originalCreateComment, string expectedCreateComment, Encoding encoding,
+            string originalUpdateComment, string expectedUpdateComment)
+        {
+            using var ms = new MemoryStream();
 
-            // Create with no encoding to autoselect it if one of the two strings is unicode
-            using (var zip = new ZipArchive(testStream, ZipArchiveMode.Create, leaveOpen: true))
+            using (var zip = new ZipArchive(ms, ZipArchiveMode.Create, leaveOpen: true, encoding))
             {
                 ZipArchiveEntry entry = zip.CreateEntry(entryName, CompressionLevel.NoCompression);
-                entry.Comment = comment;
-
-                Assert.Equal(entryName, entry.FullName);
-                Assert.Equal(comment, entry.Comment);
+                entry.Comment = originalCreateComment;
+                Assert.Equal(expectedCreateComment, entry.Comment);
             }
 
-            // Open with no encoding to verify
-            using (var zip = new ZipArchive(testStream, ZipArchiveMode.Read, leaveOpen: true))
+            using (var zip = new ZipArchive(ms, ZipArchiveMode.Read, leaveOpen: true, encoding))
             {
-                ZipArchiveEntry entry = zip.Entries.FirstOrDefault();
-
-                Assert.Equal(entryName, entry.FullName);
-                Assert.Equal(comment, entry.Comment);
+                foreach (ZipArchiveEntry entry in zip.Entries)
+                {
+                    Assert.Equal(expectedCreateComment, entry.Comment);
+                }
             }
```

Comment:
Call `Create_Comment_EntryName_Encoding_Internal` instead to avoid code duplication.

Reply:
Or even better, you could test ZipArchiveMode.Update inside `Create_Comment_EntryName_Encoding_Internal` and remove zip_UpdateTests.Comments.cs entirely.